File: Habitica/src/com/habitrpg/android/habitica/userpicture/BitmapUtils.java
package com.habitrpg.android.habitica.userpicture;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.Environment;

import java.io.File;
import java.io.FileOutputStream;

public class BitmapUtils {
    public static String getSavePath() {
        String path;
        if (hasSDCard()) {
            path = Environment.getExternalStorageDirectory().getAbsolutePath();
        } else {
            path = Environment.getDownloadCacheDirectory().getAbsolutePath();
        }
        return path+"/HabiticaImageCache";
    }

    public static Bitmap loadFromFile(String filename) {
        try {
            filename = getSavePath() +"/"+ filename;

            File f = new File(filename);
            if (!f.exists()) { return null; }
            Bitmap tmp = BitmapFactory.decodeFile(filename);
            return tmp;
        } catch (Exception e) {
            return null;
        }
    }

    public static void saveToFile(String filename,Bitmap bmp) {
        try {
            File myDir = new File(getSavePath());
            boolean res = myDir.mkdirs();

            filename = getSavePath() +"/"+ filename;

            FileOutputStream out = new FileOutputStream(filename);
            bmp.compress(Bitmap.CompressFormat.PNG, 100, out);
            out.flush();
            out.close();
        } catch(Exception e) {}
    }

    public static boolean hasSDCard() {
        String status = Environment.getExternalStorageState();
        return status.equals(Environment.MEDIA_MOUNTED);
    }
}

File: Habitica/src/main/java/com/habitrpg/android/habitica/userpicture/BitmapUtils.kt
package com.habitrpg.android.habitica.userpicture

import android.graphics.Bitmap
import java.io.File
import java.io.FileOutputStream

object BitmapUtils {
    fun clearDirectoryContent(directory: String) {
        val dirFile = File(directory)
        dirFile.listFiles()?.forEach { it.delete() }
    }

    fun saveToShareableFile(directory: String, filename: String, bmp: Bitmap): File? {
        var name = filename
        try {
            name = "$directory/$name"
            val dirFile = File(directory)
            if (!dirFile.exists()) {
                dirFile.mkdir()
            }

            val out = FileOutputStream(name)
            bmp.compress(Bitmap.CompressFormat.PNG, 100, out)
            out.flush()
            out.close()
            return File(name)
        } catch (ignored: Exception) {
        }

        return null
    }
}
